articles/views.py

```diff
@@ -55,3 +63,32 @@ def form_valid(self, form):
         """
         self.object = form.save(self.request.user)
         return HttpResponseRedirect(self.get_success_url(), status=303)
+
+
+class FavoriteArticle(IsAuthenticatedOrReadOnly, UpdateView):
```

Comment:
Did you see the `LoginRequiredMixin`? I will redirect the user back to the right page after they login.